## Непрерывные случайные величины. Функция распределения и плотность распределения вероятностей. Равномерное и нормальное распределение. Центральная предельная теорема

In [55]:
from sympy import *
from scipy.stats import norm
import numpy as np

### Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800]. Найдите ее среднее значение и дисперсию.

$$ При~ равномерном~ распределении~ математическое~ ожидание~ находится~ по~ формуле~ M(X)~ = \frac{a+b}{2}~$$

$$ Дисперсю~ найдем~ по~ формуле~ D(X)~ = \frac{(b-a)^{2}}{12} $$

In [9]:
a = 200
b = 800

In [10]:
M = (a+b)/2

In [11]:
D = (b-a)**2/12

In [13]:
print(f'Математическое ожидание равно - {M}')
print(f'Дисперсия равна - {D}')

Математическое ожидание равно - 500.0
Дисперсия равна - 30000.0


### О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

Найти правую границу величины B возможно, найдя неизвестную переменную из формулы нахождения дисперсии, решив уравнение 0.2 = (b - 0.5)**2/12

In [17]:
b = Symbol('b')

In [18]:
solved_value=solve([(b-0.5)**2/12-0.2], [b])
print(solved_value)

[(-1.04919333848297,), (2.04919333848297,)]


$$ Правая~ граница~ величины~ B~ = 2.05 $$

### Непрерывная случайная величина X распределена нормально и задана плотностью распределения f(x) = (1 / (4 * sqrt(2pi))) * exp((-(x+2)**2) / 32) Найдите: а). M(X) б). D(X) в). std(X) (среднее квадратичное отклонение)

$$ Нормальное~ распределение~ —~ это~ распределение~ вероятностей~ непрерывной~ случайной~ величины~  X, ~плотность~ вероятности~ которой~ подчиняется~ формуле:~ f(x) = \frac{1}{\sigma\sqrt{2\Pi}}\epsilon^{-\frac{(x-a)^{2}}{2\sigma^{2}}} $$

Вытащим из формулы M(X), D(X), std(X)

 M(X) = a = -2

D(X) = sigma^2 = 16

std(X) = sigma = 4

### Рост взрослого населения города X имеет нормальное распределение. Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см. Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост: а). больше 182 см б). больше 190 см в). от 166 см до 190 см г). от 166 см до 182 см д). от 158 см до 190 см е). не выше 150 см или не ниже 190 см ё). не выше 150 см или не ниже 198 см ж). ниже 166 см.

In [40]:
M = 174 #математическое ожидание
S = 8 #среднее квадратичное отклонение

Найдем вероятность того, что случайным образом выбранный взрослый человек имеет рост: больше 182 см

(182-M)/S) - формула нормализации (нахождение колл-во сигм), далее вычтем из единицы расстояние от 174 до 182 и от 0 до 174

In [46]:
P1 = 1-norm.cdf((182-M)/S) - norm.cdf((0-M)/S)
P1

0.15865525393145707

In [47]:
P2 = 1-norm.cdf((190-M)/S) - norm.cdf((0-M)/S)

In [48]:
P3 = norm.cdf((190-M)/S) - norm.cdf((166-M)/S)

In [49]:
P4 = norm.cdf((182-M)/S) - norm.cdf((166 - M)/S)

In [ ]:
P5 = norm.cdf((190-M)/S) - norm.cdf((158-M)/S)

Далее сигмы будем прибавлять так как в условии оператор "или"

In [51]:
P6 = (1 - norm.cdf((190-M)/S) - norm.cdf((0-M)/S)) + (norm.cdf((150-M)/S) - norm.cdf((0-M)/S))

In [52]:
P7 = (1 - norm.cdf((198-M)/S) - norm.cdf((0-M) / S)) + (norm.cdf((150-M)/S) - norm.cdf((0-M)/S))

In [53]:
P8 = norm.cdf((166-M)/S) - norm.cdf((0-M)/S)

In [56]:
print('Вероятность того, что случайным образом выбранный взрослый человек имеет рост:')
print(f'больше 182 см: {np.round(P1, 4)}')
print(f'больше 190 см: {np.round(P2, 4)}')
print(f'от 166 см до 190 см: {np.round(P3, 4)}')
print(f'от 166 см до 182 см: {np.round(P4, 4)}')
print(f'от 158 см до 190 см: {np.round(P5, 4)}')
print(f'не выше 150 см или не ниже 190 см: {np.round(P6, 4)}')
print(f'не выше 150 см или не ниже 198 см: {np.round(P7, 4)}')
print(f'ниже 166 см.: {np.round(P8, 4)}')

Вероятность того, что случайным образом выбранный взрослый человек имеет рост:
больше 182 см: 0.1587
больше 190 см: 0.0228
от 166 см до 190 см: 0.8186
от 166 см до 182 см: 0.6827
от 158 см до 190 см: 0.9545
не выше 150 см или не ниже 190 см: 0.0241
не выше 150 см или не ниже 198 см: 0.0027
ниже 166 см.: 0.1587


### На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

$$ Будем~ решать~ данную~ задачу~ исходя~ из~ формулы~ нормализации~ \frac{b-a}{std} $$

Среднее квадратичное отклонение равно sqrt(25)

In [22]:
b = 190
a = 178
std = 5

In [25]:
Z = (b-a)/5

In [26]:
print(f'Колл-во сигм на которые отклоняется рост человека с 190 см равно - {Z}')

Колл-во сигм на которые отклоняется рост человека с 190 см равно - 2.4
